frontend/src/pages/user/admin/allCoupons/[adminCouponID].tsx

```diff
@@ -1,218 +1,205 @@
+      return;
+    }
+    if (data.type === 'percentage' && data.discount >= 100) {
+      alert('Discount should be less than 100%');
+      return;
+    }
+    if (data.discount < 0) {
+      alert('Discount should be greater than 0');
+      return;
+    }
+    // nice "end_date" backend team 🥳
+    interface INewCoupon {
+      description: string;
+      discount: number;
+      end_date: string;
+      name: string;
+      start_date: string;
+      type: 'percentage' | 'fixed' | 'shipping';
+    }
+    // I have no idea how discount get turned into string
+    const newCoupon: INewCoupon = {
+      description: data.description,
+      discount: Number(data.discount),
+      end_date: new Date(data.expire_date).toISOString(),
+      name: data.name,
+      start_date: new Date(data.start_date).toISOString(),
+      type: data.type,
+    };
+    const resp = await fetch(`/api/admin/coupon/${coupon_id}`, {
+      method: 'PATCH',
+      headers: {
+        accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(newCoupon),
+    });
+    if (!resp.ok) {
+      const response = await resp.json();
+      alert(response.message);
+    } else {
+      navigate('/admin/manageCoupons');
     }
   };
-  const deleteTag = (index: number) => {
-    //TODO: api request
-    remove(index);
+  const OnDelete = async () => {
+    const resp = await fetch(`/api/admin/coupon/${coupon_id}`, {
+      method: 'DELETE',
+      headers: {
+        accept: 'application/json',
+      },
+    });
+    if (!resp.ok) {
+      const response = await resp.json();
+      alert(response.message);
+    } else {
+      navigate('/admin/manageCoupons');
+    }
   };
 
-  //TODO: get the init value
-  // const params = useParams();
-  // const id = params.coupon_id;
-  const initData: CouponProps = {
-    id: 0,
-    type: 'percentage',
-    name: 'Coupon',
-    description: 'this is description',
-    discount: 0,
-    start_date: '2000-01-01',
-    expire_date: '2000-01-01',
-    tags: [],
-  };
-  const setInitField = (data: CouponProps) => {
-    //set the field into the initial data
-    setValue('id', data.id);
-    setValue('type', data.type);
-    setValue('name', data.name);
-    setValue('description', data.description);
-    setValue('discount', data.discount);
-    setValue('start_date', data.start_date);
-    setValue('expire_date', data.expire_date);
-    setValue('tags', data.tags);
-  };
-  // TODO: this is just prevent the inf loop, will remove latter
-  const [isInit, setIsInit] = useState(false);
-  if (!isInit) {
-    setInitField(initData);
-    setIsInit(true);
+  useEffect(() => {
+    if (initStatus === 'success') {
+      const startDate = new Date(initData.start_date);
+      const expDate = new Date(initData.expire_date);
+      reset({
+        ...initData,
+        start_date: formatDate(startDate.toISOString()),
+        expire_date: formatDate(expDate.toISOString()),
```

Comment:
i tried to use ``toLocaleDateString().replace(/\//g, '-')`` but it still need padding 0 for month and date.
so, it will be very similar to ``formatDate()`` i written. 
imo, there's no need to change into it.